### PR TITLE
Fix type hint errors on Twisted trunk

### PR DESCRIPTION
**Pull request by clokep**
See #16289

### CODE REVIEW

File: synapse/util/file_consumer.py [hunk 1]
```diff
@@ -80,9 +82,13 @@ def registerProducer(
...
-            self._reactor,
-            self._reactor.getThreadPool(),
-            self._writer,
+            # mypy seems to get confused with the chaining of ParamSpec from
+            # run_in_background to deferToThreadPool.
+            #
+            # For Twisted trunk, ignore arg-type; for Twisted release ignore unused-ignore.
+            self._reactor,  # type: ignore[arg-type,unused-ignore]
+            self._reactor.getThreadPool(),  # type: ignore[arg-type,unused-ignore]
+            self._writer,  # type: ignore[arg-type,unused-ignore]
```
Comment (clokep) [inline synapse/util/file_consumer.py:91]: I don't really like this, but I don't see another option for getting this to work?

Comment (clokep) [reply synapse/util/file_consumer.py:91]: Note that these ignores are here for Twisted release, the Twisted Trunk job ignore unused ignores.

Comment (DMRobertson) [reply synapse/util/file_consumer.py:91]: > I don't really like this, but I don't see another option for getting this to work?

Let's roll with this. (ParamSpec chaining has been a source of pain for me, too. I think it tends not to go well if you pass it into an overloaded function?)

File: tests/http/server/_base.py [hunk 2]
```diff
@@ -335,7 +335,7 @@ def __init__(self, seen_awaits: Set[Tuple[str, ...]], request_number: int):
         self._request_number = request_number
         self._seen_awaits = seen_awaits
 
-        self._original_Deferred___next__ = Deferred.__next__
+        self._original_Deferred___next__ = Deferred.__next__  # type: ignore[misc,unused-ignore]
```
Comment (DMRobertson) [inline tests/http/server/_base.py:338]: What's the misc complaining about OOI?

**Review by DMRobertson — APPROVED** [review]
Thanks for looking into this!

File: tests/http/server/_base.py (hunk 2 above)
Comment (clokep) [reply tests/http/server/_base.py:338]:
```
tests/http/server/_base.py:338: error: Access to generic instance variables via class is ambiguous  [misc]
```

Comment (clokep) [reply tests/http/server/_base.py:338]: I think it is complaining about snagging a instance level method from a class?

**Pull request merged**